feat: Icon 컴포넌트 완료 (아이콘추가는 약 1/3 완료)

KimChunsick opened the pull request:
Signed-off-by: deokwon <deokwon@deokwonui-MacBookPro.local>

Commit:
feat: Icon 컴포넌트 완료 (아이콘추가는 약 1/3 완료)

Signed-off-by: deokwon <deokwon@deokwonui-MacBookPro.local>

Commit:
feat: Icon 컴포넌트 완료 (아이콘추가는 약 1/3 완료)

Signed-off-by: deokwon <deokwon@deokwonui-MacBookPro.local>

Vallista commented on packages/core/src/components/Icon/Activity/index.tsx:
```diff
@@ -0,0 +1,22 @@
+  return (
+    <Svg
+      viewBox='0 0 24 24'
+      width={size ?? 24}
```
전반적으로 중복되는데, useIcon 같은걸로 빼서 size를 별도로 가져오면 좋을듯요.

Vallista replied:
아니면 위에서 size = 24

Vallista commented on packages/core/src/components/Icon/Activity/index.tsx:
```diff
@@ -0,0 +1,22 @@
+      viewBox='0 0 24 24'
+      width={size ?? 24}
+      height={size ?? 24}
+      stroke={color ?? theme.colors.PRIMARY.FOREGROUND}
```
color 도 

color = theme.colors.PRIMARY.FOREGROUND 했으면 될 듯

Vallista commented on packages/blog/src/pages/IconPlayground/index.tsx:
```diff
@@ -0,0 +1,14 @@
+import { Container, Activity } from 'core'
```
Activity 형태가 아니라, Modal 처럼

Icon.Activity 이렇게 하는게 맞을듯요.

Vallista replied:
이렇게하면 중복되는거 많을듯

Vallista committed:
feat: radio button 개발

Commit:
fix: 1차 리뷰 반영

fix: Icon 접근 방식 추가 지원 (Ex. <Icon.Bell />)
fix: size, color 기본 값 설정을 useIcon에서 할 수 있게 수정

Signed-off-by: deokwon <deokwon@deokwonui-MacBookPro.local>

Vallista commented on packages/core/src/components/index.ts:
```diff
@@ -7,3 +7,6 @@ export * from './Container'
 export * from './Select'
+export * from './Icon'
+export * as Radio from './Radio'
+export * from './Icon'
```
왜 두개..?

KimChunsick replied:
rebase 과정에서 실수 한 것 같습니다. 지금 수정해서 올리도록 하겠습니다.

Vallista commented on packages/core/src/components/Icon/index.tsx:
```diff
@@ -0,0 +1,3 @@
+export * from './type'
+export * from './assets'
```
./assets 를 Icon으로 설정했는데, 이게 또 필요?

KimChunsick replied:
아이콘을 따로 하나만 접근할 수 있게하는 방법도 필요하지 않을까 싶어 추가 했습니다.
<Icon.Bell />과 <Bell />을 모두 사용할 수 있게요.

Vallista replied:
ㄴㄴ .Icon.Bell 만 하셈. 
Icon으로 래핑한 이유는 이름이 중복될 가능성이 있기 때문에 래핑한 것

KimChunsick replied:
추후 이름이 겹칠 수 있겠네요 수정하겠습니다

Vallista reviewed (approved):
리뷰 두 개 고치고 머지하면 됨

Commit:
fix: 2차 리뷰 반영

fix: ./Icon을 두번 import하던 코드 수정
fix: 네이밍 중복 방지를 위해 export * from './Icon' 코드를 제거함

Signed-off-by: deokwon <deokwon@deokwonui-MacBookPro.local>
